latency.go

```diff
@@ -19,17 +19,55 @@ import (
 	"github.com/tylertreat/hdrhistogram-writer"
 )
 
+// Test Parameters
 var (
-	ServerA, ServerB string
-	TargetPubRate    int
-	MsgSize          int
-	NumPubs          int
-	TestDuration     time.Duration
-	HistFile         string
+	ServerA       string
+	ServerB       string
+	TargetPubRate int
+	MsgSize       int
+	NumPubs       int
+	TestDuration  time.Duration
+	HistFile      string
+	Secure        bool
+	TLSca         string
+	TLSkey        string
+	TLScert       string
 )
 
+var usageStr = `
+Usage: latency-tests [options]
+
+Test Options:
+    -sa <url>        ServerA (Publish) (default: nats://localhost:4222)
+    -sb <url>        ServerB (Subscribe) (default: nats://localhost:4222)
+    -sz <int>        Message size in bytes (default: 8)
+    -tr <int>        Rate in msgs/sec (default: 1000)
+    -tt <string>     Test duration (default: 5s)
+    -hist <file>     Histogram output file
+    -secure          Enable TLS without verfication (default: false)
+    -tls_ca <string> TLS Certificate CA file
+    -tls_key <file>  TLS Private Key
+    -tls_cert <file> TLS Certificate
+`
+
 func usage() {
-	log.Fatalf("Usage: latency [-sa serverA] [-sb serverB] [-sz msgSize] [-tr msgs/sec] [-tt testTime] [-hist <file>]\n")
+	log.Fatalf(usageStr + "\n")
+}
+
+// createOpts creates options to use in a NATS connection. TLS options are set as
+// specified by command line parameters
+func createOpts() []nats.Option {
```

Comment:
...created a function to build opts because gometalinter was already complaining about the cyclomatic complexity of `main()`

Reply:
I would not worry about that too much. I prefer to see them right in the connect (with newlines of course) so I know what is being set on the connection.

Reply:
Its only three. They are optional though so array probably works. Can't remember if nil option is handled but if so you could do something like.

```
var secure, ca, cert  nats.Option

if Secure {
  secure = nats.Secure()
}
if TLScert {
  tlsCert = nats.ClientCert(TLSCert, TLSkey)
}

nats.Connect(ServerA, secure, tlsCert, ..)
```

Reply:
Great suggestion, but unfortunately no dice for the nil option.  Gave it a quick test, and we'll need to use an array.  Inside `nats.Connect()` we'll need to add a check for nil elements in the `options` array and then using vars would work.

Reply:
ok, we should add that.. ;) For now LGTM